src/statcache.c

```diff
@@ -511,8 +511,22 @@ time_t stat_cache_read_updated_children(stat_cache_t *cache, const char *path, G
 }
 
 // Use atime to store the next time we want a propfind
-time_t stat_cache_next_propfind(struct stat_cache_value value) {
-    return value.st.st_atime;
+time_t stat_cache_next_propfind(struct stat_cache_value value, const char *path) {
+    static const char *funcname = "stat_cache_negative_entry";
+    // Allow propfinds in first 5 seconds of creation of negative entry,
+    // since modules often check for (non-)existence of file before creating it
+    const time_t probation_delay = 5;
+    time_t curtime;
+    curtime = time(NULL);
+    if (value.st.st_ctime + probation_delay > curtime) {
+        log_print(LOG_INFO, SECTION_FUSEDAV_STAT, "%s: still within probation period %s; ctime: %lu; probation: %lu", 
+                funcname, path, value.st.st_ctime, curtime);
+        return curtime;
```

Comment:
I'd prefer something that isn't the current time. Would it be okay to return zero in order to indicate that the "next propfind" is ASAP?

Reply:
754         else {
   755             time_t current_time = time(NULL);
   756             time_t next_time;
   757
   758             next_time = stat_cache_next_propfind(*value, path);
   759
   760             // Our "time to next propfind" is still in the future, so no propfind ...
   761             if (next_time > current_time) {

I'd prefer to return the 'time to next propfind', which is in this case, immediately.